This is the initial notebook to explore the data and run the first models. 

In [1]:
import pandas as pd
import numpy as np

from sklearn.impute import SimpleImputer
from sklearn.ensemble import RandomForestRegressor
import xgboost as xgb
from sklearn.preprocessing import OneHotEncoder

from fastai.imports import *
from fastai.tabular.all import *

from numpy import random
from sklearn.model_selection import train_test_split
from sklearn.utils import shuffle
from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_squared_log_error
from sklearn import metrics
from sklearn.model_selection import KFold, cross_val_score
from sklearn.model_selection import StratifiedKFold

import eli5
from eli5.sklearn import PermutationImportance

from scipy.stats import skew, norm
from catboost import CatBoostRegressor
import optuna
from optuna.samplers import TPESampler

import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)

In [2]:
test = pd.read_csv('test.csv')
sample = pd.read_csv('sample_submission.csv')
train = pd.read_csv('train.csv')

Based on initial analysis the train and test datasets have similar characteristics, so it will be easier to combine them for imputation and data analysis work. 

In [3]:
train_test = pd.concat([train, test], ignore_index=True)

In [4]:
pd.isnull(train_test).sum()[pd.isnull(train_test).sum() > 0]

MSZoning           4
LotFrontage      486
Alley           2721
Utilities          2
Exterior1st        1
Exterior2nd        1
MasVnrType        24
MasVnrArea        23
BsmtQual          81
BsmtCond          82
BsmtExposure      82
BsmtFinType1      79
BsmtFinSF1         1
BsmtFinType2      80
BsmtFinSF2         1
BsmtUnfSF          1
TotalBsmtSF        1
Electrical         1
BsmtFullBath       2
BsmtHalfBath       2
KitchenQual        1
Functional         2
FireplaceQu     1420
GarageType       157
GarageYrBlt      159
GarageFinish     159
GarageCars         1
GarageArea         1
GarageQual       159
GarageCond       159
PoolQC          2909
Fence           2348
MiscFeature     2814
SaleType           1
SalePrice       1459
dtype: int64

It looks like Alley, FireplaceQu, PoolQC, Fence and MiscFeature have significant numbers of missing data. So it will be best to eliminate those rows. There are a number of rows that have less than 5 rows with missing data. Since some of these are categorical and some are continuous data, their missing data will be replaced with the most frequent value. 

In [5]:
drop_high_nan=['Alley','FireplaceQu','PoolQC','Fence','MiscFeature']
train_test=train_test.drop(drop_high_nan,axis=1)
small_nan_cols = ['MSZoning', 'Utilities', 'Exterior1st', 'Exterior2nd', 'BsmtFinSF1', 'BsmtFinSF2', 'BsmtUnfSF', 
                  'TotalBsmtSF', 'Electrical', 'BsmtFullBath', 'BsmtHalfBath', 'KitchenQual', 'Functional', 'GarageCars', 
                  'GarageArea','SaleType', 'SaleCondition']
small_impute = SimpleImputer(strategy='most_frequent')
train_test[small_nan_cols] = pd.DataFrame(small_impute.fit_transform(train_test[small_nan_cols]),columns=small_nan_cols)

The following columns seem to have one value significantly larger than the rest, and it would probably be best to use the mode, or most common, value to feel each NaN value: MasVnrType, MasVnrArea, BsmtCond, BsmtExposure, BsmtFinType2, GarageType, GarageFinish, GarageQual, and GarageCond. That is represents 9 out of the 13 columns. 

BsmtQual has two values larger than the rest: Gd and TA. But it only has 2.8% NaNs, so simply using the mode might be good enough.  

GarageYrBlt has 59 NaNs out 2919 rows which is only 2%. It has a dispersed set of values, so it might be easiest just to have any NaNs have the same value as YearBuilt. 

BsmtFinType1 has only 2.7% value of NaNs, and most two of its largest values are GLQ and Unf. It might be easiest to use the mode here. 

LotFrontage has 486 NaNs out of 2919 rows which is a pretty high 16.7%. It has a dispersed range of values, but looking at its characteristics from the describe function above, it seems to have a pretty even distribution with a mean of 10,168 and a median of 9,453. So using the mean to fill in the NaNs seems reasonable. If it turns out this value has a high impact on the predictability of the SalePrice, then it might be good to revisit this assumption. 

In [6]:
mode_cols = ['MasVnrType', 'MasVnrArea', 'BsmtCond', 'BsmtExposure', 'BsmtFinType2', 'GarageType', 'GarageFinish', 
             'GarageQual','GarageCond', 'BsmtQual', 'BsmtFinType1']
mode_impute = SimpleImputer(strategy='most_frequent')
train_test[mode_cols] = pd.DataFrame(mode_impute.fit_transform(train_test[mode_cols]),columns=mode_cols)
train_test['LotFrontage'].fillna((train_test['LotFrontage'].mean()), inplace=True)
train_test['GarageYrBlt'] = train_test['GarageYrBlt'].fillna(train_test['YearBuilt'])

In [7]:
pd.isnull(train_test).sum()[pd.isnull(train_test).sum() > 0]

SalePrice    1459
dtype: int64

In [8]:
train_test['BsmtQual'].unique()

array(['Gd', 'TA', 'Ex', 'Fa'], dtype=object)

In [9]:
train_test['BsmtFinType1'].unique()

array(['GLQ', 'ALQ', 'Unf', 'Rec', 'BLQ', 'LwQ'], dtype=object)

In [10]:
train_test.BsmtQual = train_test.BsmtQual.replace({"Ex": 110, "Gd": 95, "TA": 85, "Fa": 75, "Po": 60, "NA": 0})

In [11]:
train_test.BsmtFinType1 = train_test.BsmtFinType1.replace({"GLQ": 6, "ALQ": 5, "BLQ": 4, "Rec": 3, "LwQ": 2, "Unf": 1,
                                                         "NA": 0})

It will be necessary to identify all the columns that have non-numeric object values and then convert them to numeric values. 

In [12]:
obj_cols = list(train_test.select_dtypes(['object']).columns)
obj_cols

['MSZoning',
 'Street',
 'LotShape',
 'LandContour',
 'Utilities',
 'LotConfig',
 'LandSlope',
 'Neighborhood',
 'Condition1',
 'Condition2',
 'BldgType',
 'HouseStyle',
 'RoofStyle',
 'RoofMatl',
 'Exterior1st',
 'Exterior2nd',
 'MasVnrType',
 'MasVnrArea',
 'ExterQual',
 'ExterCond',
 'Foundation',
 'BsmtCond',
 'BsmtExposure',
 'BsmtFinSF1',
 'BsmtFinType2',
 'BsmtFinSF2',
 'BsmtUnfSF',
 'TotalBsmtSF',
 'Heating',
 'HeatingQC',
 'CentralAir',
 'Electrical',
 'BsmtFullBath',
 'BsmtHalfBath',
 'KitchenQual',
 'Functional',
 'GarageType',
 'GarageFinish',
 'GarageCars',
 'GarageArea',
 'GarageQual',
 'GarageCond',
 'PavedDrive',
 'SaleType',
 'SaleCondition']

In [13]:
for column in obj_cols:
     train_test[column] = pd.factorize(train_test[column], sort=True)[0]

To create some new columns that might compound the effects of the existing columns with higher impact on the xgb score. 

In [14]:
train_test['QualCondSum'] = train_test['OverallQual'] + train_test['OverallCond']
train_test['RemodTime'] = train_test['YearRemodAdd'] - train_test['YearBuilt']
train_test['BsmtFinTypeSF1'] = train_test['BsmtFinType1'] * train_test['BsmtFinSF1']
train_test['TotalFlrSF'] = train_test['1stFlrSF'] + train_test['2ndFlrSF']
train_test['TotalFinSF'] = train_test['GrLivArea'] + train_test['BsmtFinSF1']
train_test['GarageCarArea'] = train_test['GarageArea'] * train_test['GarageCars']
train_test['TotalSF'] = train_test['1stFlrSF'] + train_test['2ndFlrSF'] + train_test['TotalBsmtSF']

To now eliminate the columns that have negative or zero influence on the xgb score. 

In [15]:
train_test.drop(['Utilities','Street','TotRmsAbvGrd','3SsnPorch','YrSold','Exterior2nd','LotConfig',
                              'HouseStyle','EnclosedPorch','WoodDeckSF','Foundation','RoofMatl','Electrical',
                              'GarageType','LotFrontage','SaleType','MoSold','BsmtExposure','1stFlrSF',
                              'BsmtFinSF1','Exterior1st','KitchenQual','TotalFlrSF'], axis=1, inplace=True)

To look at how many features have a skew above 0.5, since high skew can be an issue in regression analysis. 

In [16]:
skew_features = train_test.apply(lambda x: skew(x)).sort_values(ascending=False)

In [17]:
high_skew = skew_features[skew_features > 0.5]
skew_index = high_skew.index

In [18]:
skew_index

Index(['MiscVal', 'PoolArea', 'LotArea', 'LowQualFinSF', 'Heating',
       'Condition2', 'LandSlope', 'KitchenAbvGr', 'ScreenPorch',
       'BsmtHalfBath', 'BsmtFinSF2', 'Condition1', 'OpenPorchSF', 'BldgType',
       'RemodTime', 'MasVnrArea', 'RoofStyle', 'MSSubClass', 'GarageCarArea',
       'GrLivArea', 'TotalFinSF', 'BsmtFinTypeSF1', 'TotalSF', '2ndFlrSF',
       'BsmtQual', 'Fireplaces', 'HalfBath', 'BsmtFullBath', 'OverallCond'],
      dtype='object')

In [19]:
print("There are {} numerical features with Skew > 0.5 :".format(high_skew.shape[0]))
skewness = pd.DataFrame({'Skew' :high_skew})
skew_features.head(30)

There are 29 numerical features with Skew > 0.5 :


MiscVal           21.947195
PoolArea          16.898328
LotArea           12.822431
LowQualFinSF      12.088761
Heating           12.078788
Condition2        12.060093
LandSlope          4.975157
KitchenAbvGr       4.302254
ScreenPorch        3.946694
BsmtHalfBath       3.931594
BsmtFinSF2         3.476562
Condition1         2.983114
OpenPorchSF        2.535114
BldgType           2.192261
RemodTime          2.063712
MasVnrArea         1.556310
RoofStyle          1.553307
MSSubClass         1.375457
GarageCarArea      1.296672
GrLivArea          1.269358
TotalFinSF         1.159630
BsmtFinTypeSF1     0.932490
TotalSF            0.922771
2ndFlrSF           0.861675
BsmtQual           0.796302
Fireplaces         0.733495
HalfBath           0.694566
BsmtFullBath       0.624832
OverallCond        0.570312
HeatingQC          0.486656
dtype: float64

To normalize the features with skew above 0.5.

In [20]:
for i in skew_index:
    train_test[i] = np.log1p(train_test[i])

C:\Users\tompe\anaconda3\lib\site-packages\pandas\core\arraylike.py:397: RuntimeWarning: divide by zero encountered in log1p
  result = getattr(ufunc, method)(*inputs, **kwargs)


To create a column with the log of the SalePrice to match the evaluation process in the contest. 

In [21]:
train_test['LogSalePrice'] = train_test['SalePrice'].apply(np.log)

To separate the train_test dataset back into the train and test datasets, identify the independent and dependent columns, and create the validation split.

In [22]:
train = train_test[train_test['SalePrice'].notnull()].copy()
test = train_test[train_test['SalePrice'].isnull()].drop(['SalePrice','LogSalePrice'],axis=1)
X = train.drop(['SalePrice','LogSalePrice'],axis=1)
y = train.LogSalePrice

In [23]:
#X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [24]:
'''def catboost_objective(trial):
    param = {  
        
        "colsample_bylevel": trial.suggest_float("colsample_bylevel", 0.01, 0.1),
        "learning_rate": trial.suggest_float("learning_rate", .001, 0.1),
        "depth": trial.suggest_int("depth", 1, 12),
        "l2_leaf_reg":trial.suggest_float("l2_leaf_reg", .0001, 25.0),
        "subsample": trial.suggest_float("subsample", 0.05, 1.0),
        "bagging_temperature":trial.suggest_float("bagging_temperature", 0, 10),
        "model_size_reg":trial.suggest_float("model_size_reg", 0.1, 1.0, log=True),
        "boosting_type": trial.suggest_categorical("boosting_type", ["Ordered", "Plain"]),
            }        

    
    model = CatBoostRegressor(**param)
    model.fit(X_train, y_train, eval_set=[(X_test, y_test)], verbose=0)
    y_pred = model.predict(X_test)
    rmse = np.sqrt(mean_squared_error(y_test, y_pred))
    return rmse'''

In [25]:
'''study_catboost = optuna.create_study(sampler=TPESampler(), direction='minimize')
study_catboost.optimize(catboost_objective, n_trials=1000)
print('Number of finished trials:', len(study_catboost.trials))
print('Best trial:', study_catboost.best_trial.params)'''

[I 2023-06-15 14:25:42,012] A new study created in memory with name: no-name-54afe4c3-da6e-4fac-b65e-e290ba89ad30
[I 2023-06-15 14:25:44,032] Trial 0 finished with value: 0.12696314120477328 and parameters: {'colsample_bylevel': 0.05582062739495187, 'learning_rate': 0.07916394845959324, 'depth': 3, 'l2_leaf_reg': 0.9589700063681216, 'subsample': 0.8525357623798345, 'bagging_temperature': 8.368103410643174, 'model_size_reg': 0.9209099514553063, 'boosting_type': 'Ordered'}. Best is trial 0 with value: 0.12696314120477328.
[I 2023-06-15 14:25:45,538] Trial 1 finished with value: 0.13586449265732786 and parameters: {'colsample_bylevel': 0.019508114504635833, 'learning_rate': 0.06558454091170078, 'depth': 7, 'l2_leaf_reg': 9.926353604347035, 'subsample': 0.9736287014087781, 'bagging_temperature': 9.141293097691506, 'model_size_reg': 0.37781974235158305, 'boosting_type': 'Ordered'}. Best is trial 0 with value: 0.12696314120477328.
[I 2023-06-15 14:25:46,754] Trial 2 finished with value: 0.15

[I 2023-06-15 14:27:03,533] Trial 20 finished with value: 0.13475009792756254 and parameters: {'colsample_bylevel': 0.07458773529884201, 'learning_rate': 0.07392659876588242, 'depth': 9, 'l2_leaf_reg': 3.6978153327637773, 'subsample': 0.7347541654219124, 'bagging_temperature': 5.081457166726072, 'model_size_reg': 0.9855159901641836, 'boosting_type': 'Ordered'}. Best is trial 0 with value: 0.12696314120477328.
[I 2023-06-15 14:27:05,280] Trial 21 finished with value: 0.13007580445125536 and parameters: {'colsample_bylevel': 0.059838912744675246, 'learning_rate': 0.08583263472114985, 'depth': 5, 'l2_leaf_reg': 3.5388202773159803, 'subsample': 0.8848783528678854, 'bagging_temperature': 2.7127964788021894, 'model_size_reg': 0.7899913641345258, 'boosting_type': 'Plain'}. Best is trial 0 with value: 0.12696314120477328.
[I 2023-06-15 14:27:07,483] Trial 22 finished with value: 0.13321873725063804 and parameters: {'colsample_bylevel': 0.060742454157418566, 'learning_rate': 0.08917782737395967

[I 2023-06-15 14:28:19,933] Trial 40 finished with value: 0.13998590645032774 and parameters: {'colsample_bylevel': 0.024910357333262134, 'learning_rate': 0.08592970702022863, 'depth': 1, 'l2_leaf_reg': 5.2338394932292065, 'subsample': 0.6505161438599784, 'bagging_temperature': 8.58573143645042, 'model_size_reg': 0.8613004064981176, 'boosting_type': 'Plain'}. Best is trial 0 with value: 0.12696314120477328.
[I 2023-06-15 14:28:21,220] Trial 41 finished with value: 0.14244167412620018 and parameters: {'colsample_bylevel': 0.010783294971064908, 'learning_rate': 0.07010212660987165, 'depth': 3, 'l2_leaf_reg': 3.8480123143076765, 'subsample': 0.7442860772919991, 'bagging_temperature': 4.930913748456754, 'model_size_reg': 0.8728384490321842, 'boosting_type': 'Plain'}. Best is trial 0 with value: 0.12696314120477328.
[I 2023-06-15 14:28:22,470] Trial 42 finished with value: 0.13021400627524762 and parameters: {'colsample_bylevel': 0.041261815526692024, 'learning_rate': 0.08201336916728423, '

[I 2023-06-15 14:28:53,248] Trial 60 finished with value: 0.1303631635662064 and parameters: {'colsample_bylevel': 0.05308407222870424, 'learning_rate': 0.06700805385424385, 'depth': 3, 'l2_leaf_reg': 0.8891754111240953, 'subsample': 0.7465562155885814, 'bagging_temperature': 7.126939204770908, 'model_size_reg': 0.7948990938802735, 'boosting_type': 'Plain'}. Best is trial 55 with value: 0.12510926896141228.
[I 2023-06-15 14:28:54,386] Trial 61 finished with value: 0.13595700373148945 and parameters: {'colsample_bylevel': 0.03773601553381137, 'learning_rate': 0.08392031831244287, 'depth': 2, 'l2_leaf_reg': 2.306017984089096, 'subsample': 0.6556331161160145, 'bagging_temperature': 4.905638919012867, 'model_size_reg': 0.9211433580158694, 'boosting_type': 'Plain'}. Best is trial 55 with value: 0.12510926896141228.
[I 2023-06-15 14:28:55,884] Trial 62 finished with value: 0.12735243079588474 and parameters: {'colsample_bylevel': 0.043716140846883966, 'learning_rate': 0.0868471972825816, 'de

[I 2023-06-15 14:29:51,148] Trial 80 finished with value: 0.1310378763144958 and parameters: {'colsample_bylevel': 0.039666500098139275, 'learning_rate': 0.09318604932655852, 'depth': 2, 'l2_leaf_reg': 2.4920731572484707, 'subsample': 0.8246205707859691, 'bagging_temperature': 6.26152728000392, 'model_size_reg': 0.8748639647837457, 'boosting_type': 'Ordered'}. Best is trial 55 with value: 0.12510926896141228.
[I 2023-06-15 14:29:52,548] Trial 81 finished with value: 0.12899278932792196 and parameters: {'colsample_bylevel': 0.04538089601576495, 'learning_rate': 0.0970452173726886, 'depth': 2, 'l2_leaf_reg': 0.585347493859919, 'subsample': 0.7038895719180646, 'bagging_temperature': 4.686384919758386, 'model_size_reg': 0.9554405658606617, 'boosting_type': 'Plain'}. Best is trial 55 with value: 0.12510926896141228.
[I 2023-06-15 14:29:53,806] Trial 82 finished with value: 0.13583153850000076 and parameters: {'colsample_bylevel': 0.047699593669747915, 'learning_rate': 0.08082869138883618, '

[I 2023-06-15 14:30:25,712] Trial 100 finished with value: 0.1344970567836158 and parameters: {'colsample_bylevel': 0.049815338160118766, 'learning_rate': 0.08278043630883733, 'depth': 4, 'l2_leaf_reg': 0.05135511267979076, 'subsample': 0.7216663635384577, 'bagging_temperature': 6.416288073471819, 'model_size_reg': 0.7517581115725658, 'boosting_type': 'Plain'}. Best is trial 55 with value: 0.12510926896141228.
[I 2023-06-15 14:30:27,570] Trial 101 finished with value: 0.12468710732111159 and parameters: {'colsample_bylevel': 0.04653324234869746, 'learning_rate': 0.08668845154398479, 'depth': 3, 'l2_leaf_reg': 0.8625668417338674, 'subsample': 0.6737109788086397, 'bagging_temperature': 4.597588841210365, 'model_size_reg': 0.9537787027888864, 'boosting_type': 'Plain'}. Best is trial 101 with value: 0.12468710732111159.
[I 2023-06-15 14:30:28,861] Trial 102 finished with value: 0.13079397247658095 and parameters: {'colsample_bylevel': 0.0273581735011266, 'learning_rate': 0.0870521658015592

[I 2023-06-15 14:31:00,376] Trial 120 finished with value: 0.13212437541243743 and parameters: {'colsample_bylevel': 0.05067446751386998, 'learning_rate': 0.09961033466816241, 'depth': 7, 'l2_leaf_reg': 4.821217403389558, 'subsample': 0.7825643193966805, 'bagging_temperature': 2.4589639767583753, 'model_size_reg': 0.8709114260388825, 'boosting_type': 'Plain'}. Best is trial 101 with value: 0.12468710732111159.
[I 2023-06-15 14:31:01,416] Trial 121 finished with value: 0.13207663129383523 and parameters: {'colsample_bylevel': 0.044077390685602624, 'learning_rate': 0.09275491557483143, 'depth': 1, 'l2_leaf_reg': 2.892707907205974, 'subsample': 0.7226518078989886, 'bagging_temperature': 5.085282923908852, 'model_size_reg': 0.799736672652847, 'boosting_type': 'Plain'}. Best is trial 101 with value: 0.12468710732111159.
[I 2023-06-15 14:31:02,823] Trial 122 finished with value: 0.13113276018965947 and parameters: {'colsample_bylevel': 0.04597585487426426, 'learning_rate': 0.0992792611134632

[I 2023-06-15 14:31:35,297] Trial 140 finished with value: 0.1277028246613056 and parameters: {'colsample_bylevel': 0.051979895663073206, 'learning_rate': 0.0988474709152204, 'depth': 3, 'l2_leaf_reg': 3.5631753408402322, 'subsample': 0.6615552335848235, 'bagging_temperature': 4.33578681614339, 'model_size_reg': 0.8864043742413563, 'boosting_type': 'Plain'}. Best is trial 101 with value: 0.12468710732111159.
[I 2023-06-15 14:31:37,226] Trial 141 finished with value: 0.12805220580396803 and parameters: {'colsample_bylevel': 0.04832390438203048, 'learning_rate': 0.09590264129486473, 'depth': 3, 'l2_leaf_reg': 3.028060009525951, 'subsample': 0.690637807401729, 'bagging_temperature': 4.507803130938137, 'model_size_reg': 0.9532691267016955, 'boosting_type': 'Plain'}. Best is trial 101 with value: 0.12468710732111159.
[I 2023-06-15 14:31:39,246] Trial 142 finished with value: 0.12906891017932115 and parameters: {'colsample_bylevel': 0.045340043439788424, 'learning_rate': 0.08613949336794034,

[I 2023-06-15 14:32:18,455] Trial 160 finished with value: 0.12753784679425442 and parameters: {'colsample_bylevel': 0.055744841063504864, 'learning_rate': 0.09219737594369232, 'depth': 2, 'l2_leaf_reg': 5.163384717796955, 'subsample': 0.7300392836867629, 'bagging_temperature': 8.117405039731448, 'model_size_reg': 0.9963631742669525, 'boosting_type': 'Plain'}. Best is trial 101 with value: 0.12468710732111159.
[I 2023-06-15 14:32:19,802] Trial 161 finished with value: 0.131680553661222 and parameters: {'colsample_bylevel': 0.0541530962865763, 'learning_rate': 0.09821497431133455, 'depth': 3, 'l2_leaf_reg': 3.8057943740675975, 'subsample': 0.6641147365874984, 'bagging_temperature': 4.543369199413906, 'model_size_reg': 0.8927674989937429, 'boosting_type': 'Plain'}. Best is trial 101 with value: 0.12468710732111159.
[I 2023-06-15 14:32:21,294] Trial 162 finished with value: 0.12909297113099683 and parameters: {'colsample_bylevel': 0.04991487455075677, 'learning_rate': 0.09577422167486144,

[I 2023-06-15 14:32:47,784] Trial 180 finished with value: 0.12684130680040795 and parameters: {'colsample_bylevel': 0.05922873909682909, 'learning_rate': 0.08663239819436352, 'depth': 2, 'l2_leaf_reg': 10.765910347590886, 'subsample': 0.7598667337185405, 'bagging_temperature': 9.105523717296892, 'model_size_reg': 0.31895550048665244, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:32:49,272] Trial 181 finished with value: 0.12618076287867333 and parameters: {'colsample_bylevel': 0.06126096693251796, 'learning_rate': 0.08415326527901723, 'depth': 2, 'l2_leaf_reg': 4.617557253616525, 'subsample': 0.7538634668172264, 'bagging_temperature': 9.209355803201246, 'model_size_reg': 0.324464987937686, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:32:50,589] Trial 182 finished with value: 0.12699874054222018 and parameters: {'colsample_bylevel': 0.061166681691219645, 'learning_rate': 0.08529436249634108

[I 2023-06-15 14:33:29,445] Trial 200 finished with value: 0.1278119754832308 and parameters: {'colsample_bylevel': 0.06234792214673324, 'learning_rate': 0.0692901149109053, 'depth': 2, 'l2_leaf_reg': 5.1556062041757444, 'subsample': 0.7363707878145969, 'bagging_temperature': 9.646857644376128, 'model_size_reg': 0.36431702273256783, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:33:30,623] Trial 201 finished with value: 0.12548561567573954 and parameters: {'colsample_bylevel': 0.05844957577525935, 'learning_rate': 0.0766296640532526, 'depth': 2, 'l2_leaf_reg': 4.840366641541917, 'subsample': 0.6951115274256606, 'bagging_temperature': 9.21492159049244, 'model_size_reg': 0.33365578441179855, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:33:31,994] Trial 202 finished with value: 0.12616938105505018 and parameters: {'colsample_bylevel': 0.05845306009694382, 'learning_rate': 0.07639562743488962, '

[I 2023-06-15 14:33:59,633] Trial 220 finished with value: 0.13691854227016922 and parameters: {'colsample_bylevel': 0.05708814450784333, 'learning_rate': 0.07095408878859083, 'depth': 9, 'l2_leaf_reg': 6.741346185836446, 'subsample': 0.6747431397223785, 'bagging_temperature': 7.776472614223799, 'model_size_reg': 0.30462821637415727, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:34:00,666] Trial 221 finished with value: 0.1284399580714201 and parameters: {'colsample_bylevel': 0.059018672455420355, 'learning_rate': 0.07690754545379505, 'depth': 2, 'l2_leaf_reg': 4.728204065889255, 'subsample': 0.694622187005013, 'bagging_temperature': 9.221213440590347, 'model_size_reg': 0.34825286268451494, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:34:01,766] Trial 222 finished with value: 0.12621547062990582 and parameters: {'colsample_bylevel': 0.05835658192568008, 'learning_rate': 0.07573513996345263,

[I 2023-06-15 14:34:21,715] Trial 240 finished with value: 0.13344615321986125 and parameters: {'colsample_bylevel': 0.04231004063201583, 'learning_rate': 0.08010021039755343, 'depth': 6, 'l2_leaf_reg': 3.1935358081585496, 'subsample': 0.6682481776401937, 'bagging_temperature': 3.276993017224169, 'model_size_reg': 0.4825459755945388, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:34:22,836] Trial 241 finished with value: 0.12475542249701485 and parameters: {'colsample_bylevel': 0.05799352239848096, 'learning_rate': 0.07654478018362557, 'depth': 2, 'l2_leaf_reg': 4.614411902058932, 'subsample': 0.7027557968598984, 'bagging_temperature': 9.230167045188038, 'model_size_reg': 0.362673103002188, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:34:23,891] Trial 242 finished with value: 0.1260669964967488 and parameters: {'colsample_bylevel': 0.05776491190889397, 'learning_rate': 0.07679664867498981, '

[I 2023-06-15 14:34:41,865] Trial 260 finished with value: 0.135564886186801 and parameters: {'colsample_bylevel': 0.018961621407615255, 'learning_rate': 0.08680779615762949, 'depth': 2, 'l2_leaf_reg': 3.93947219447358, 'subsample': 0.5219858028722021, 'bagging_temperature': 3.647283275957182, 'model_size_reg': 0.3683414546533666, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:34:42,767] Trial 261 finished with value: 0.12631893598085767 and parameters: {'colsample_bylevel': 0.05820471325947754, 'learning_rate': 0.07201831282852422, 'depth': 2, 'l2_leaf_reg': 4.453687150997755, 'subsample': 0.17531350127516293, 'bagging_temperature': 9.696379012394274, 'model_size_reg': 0.45670843722310234, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:34:43,769] Trial 262 finished with value: 0.1284852404015601 and parameters: {'colsample_bylevel': 0.06088175652084922, 'learning_rate': 0.07715813997487181, '

[I 2023-06-15 14:35:03,193] Trial 280 finished with value: 0.13094788612732755 and parameters: {'colsample_bylevel': 0.0748299467904831, 'learning_rate': 0.04459190092275721, 'depth': 2, 'l2_leaf_reg': 5.306047114269981, 'subsample': 0.9635616790242852, 'bagging_temperature': 8.567897272700492, 'model_size_reg': 0.29984665707034985, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:35:04,466] Trial 281 finished with value: 0.1287953945579353 and parameters: {'colsample_bylevel': 0.04020983544285343, 'learning_rate': 0.05902070233906749, 'depth': 3, 'l2_leaf_reg': 5.764347606510564, 'subsample': 0.7208397880117324, 'bagging_temperature': 3.527755027610808, 'model_size_reg': 0.4856757211060154, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:35:05,537] Trial 282 finished with value: 0.16228314966337462 and parameters: {'colsample_bylevel': 0.04341009966008014, 'learning_rate': 0.008564535155920393, 

[I 2023-06-15 14:35:28,892] Trial 300 finished with value: 0.1288933880119682 and parameters: {'colsample_bylevel': 0.05538367869197585, 'learning_rate': 0.0912464837639296, 'depth': 2, 'l2_leaf_reg': 3.6590802444185813, 'subsample': 0.6822601708107172, 'bagging_temperature': 9.181339788010376, 'model_size_reg': 0.29326329069466667, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:35:34,524] Trial 301 finished with value: 0.13559995256041482 and parameters: {'colsample_bylevel': 0.04890991495690188, 'learning_rate': 0.09015743440838406, 'depth': 9, 'l2_leaf_reg': 2.980522330921237, 'subsample': 0.9326818867638329, 'bagging_temperature': 3.102798619171828, 'model_size_reg': 0.5226542369170539, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:35:35,378] Trial 302 finished with value: 0.13361030510766167 and parameters: {'colsample_bylevel': 0.09240140835955626, 'learning_rate': 0.09264832286808947, 

[I 2023-06-15 14:35:58,417] Trial 320 finished with value: 0.12967935511390954 and parameters: {'colsample_bylevel': 0.04528280245809824, 'learning_rate': 0.08616532631379163, 'depth': 3, 'l2_leaf_reg': 6.9388817379949135, 'subsample': 0.3285379375135361, 'bagging_temperature': 8.088667311301526, 'model_size_reg': 0.35161923125192096, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:35:59,344] Trial 321 finished with value: 0.13772455561057925 and parameters: {'colsample_bylevel': 0.01309476883766222, 'learning_rate': 0.07511375710103237, 'depth': 2, 'l2_leaf_reg': 1.9370501062791008, 'subsample': 0.6536164142472554, 'bagging_temperature': 4.91413541617353, 'model_size_reg': 0.835837362728047, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:36:00,453] Trial 322 finished with value: 0.13277128450563055 and parameters: {'colsample_bylevel': 0.07391969329627349, 'learning_rate': 0.08921563772964891,

[I 2023-06-15 14:36:29,165] Trial 340 finished with value: 0.1328045262850885 and parameters: {'colsample_bylevel': 0.06341317762481287, 'learning_rate': 0.09003587472323034, 'depth': 2, 'l2_leaf_reg': 7.2005860160130855, 'subsample': 0.7043826088215065, 'bagging_temperature': 2.6018451690510953, 'model_size_reg': 0.49825038992920806, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:36:30,648] Trial 341 finished with value: 0.12780899715827626 and parameters: {'colsample_bylevel': 0.054948476257489304, 'learning_rate': 0.07004492867981774, 'depth': 3, 'l2_leaf_reg': 11.899094542136881, 'subsample': 0.7613957298843076, 'bagging_temperature': 5.4504297169942335, 'model_size_reg': 0.34303929365261504, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:36:31,982] Trial 342 finished with value: 0.12663298852365695 and parameters: {'colsample_bylevel': 0.04469123850291977, 'learning_rate': 0.0871033108530

[I 2023-06-15 14:37:06,387] Trial 360 finished with value: 0.12647750752460082 and parameters: {'colsample_bylevel': 0.09506311765182282, 'learning_rate': 0.08176636027776295, 'depth': 2, 'l2_leaf_reg': 0.9198422583871316, 'subsample': 0.7229422951484661, 'bagging_temperature': 7.478866371276801, 'model_size_reg': 0.5306978023236707, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:37:09,820] Trial 361 finished with value: 0.1327991110511896 and parameters: {'colsample_bylevel': 0.09770432726647252, 'learning_rate': 0.08601970264720837, 'depth': 6, 'l2_leaf_reg': 7.875805836293512, 'subsample': 0.6819521337822323, 'bagging_temperature': 3.2237914906285776, 'model_size_reg': 0.381090276303644, 'boosting_type': 'Plain'}. Best is trial 167 with value: 0.1237078717517025.
[I 2023-06-15 14:37:11,193] Trial 362 finished with value: 0.12863701080218468 and parameters: {'colsample_bylevel': 0.08750422318148667, 'learning_rate': 0.06632469830211794, 

[I 2023-06-15 14:37:41,101] Trial 380 finished with value: 0.129969763916641 and parameters: {'colsample_bylevel': 0.08298145896869755, 'learning_rate': 0.050543402513348545, 'depth': 2, 'l2_leaf_reg': 6.102037668032619, 'subsample': 0.7629018938697034, 'bagging_temperature': 8.166316571824662, 'model_size_reg': 0.365201069494234, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:37:42,606] Trial 381 finished with value: 0.12602806089541557 and parameters: {'colsample_bylevel': 0.07598775344180124, 'learning_rate': 0.07236663232346507, 'depth': 2, 'l2_leaf_reg': 8.200985294013579, 'subsample': 0.7294520012408535, 'bagging_temperature': 9.430291474306134, 'model_size_reg': 0.5520432039998954, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:37:54,881] Trial 382 finished with value: 0.13706057961775386 and parameters: {'colsample_bylevel': 0.08027568094154773, 'learning_rate': 0.08144869777292826, 

[I 2023-06-15 14:38:17,223] Trial 400 finished with value: 0.1333129409986421 and parameters: {'colsample_bylevel': 0.03759627864908835, 'learning_rate': 0.07601126895968525, 'depth': 1, 'l2_leaf_reg': 2.6158207056695613, 'subsample': 0.9556039034482654, 'bagging_temperature': 7.779823690362547, 'model_size_reg': 0.8435780242985721, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:38:18,801] Trial 401 finished with value: 0.12792716547843208 and parameters: {'colsample_bylevel': 0.09429624359704832, 'learning_rate': 0.08868794404023489, 'depth': 3, 'l2_leaf_reg': 5.630851609054922, 'subsample': 0.4962258065018283, 'bagging_temperature': 2.7320672431250186, 'model_size_reg': 0.30438493105865894, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:38:19,933] Trial 402 finished with value: 0.13241118907653696 and parameters: {'colsample_bylevel': 0.054798497756802726, 'learning_rate': 0.04844419708403

[I 2023-06-15 14:38:41,217] Trial 420 finished with value: 0.1264827428254055 and parameters: {'colsample_bylevel': 0.081475390905291, 'learning_rate': 0.07593475202668451, 'depth': 2, 'l2_leaf_reg': 5.795225389207708, 'subsample': 0.8240336020112768, 'bagging_temperature': 5.057584061050217, 'model_size_reg': 0.23636088835411173, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:38:42,274] Trial 421 finished with value: 0.12934836735758024 and parameters: {'colsample_bylevel': 0.08400118361449585, 'learning_rate': 0.07771695818949716, 'depth': 2, 'l2_leaf_reg': 1.7246555632970442, 'subsample': 0.2800131095748881, 'bagging_temperature': 7.164519712548096, 'model_size_reg': 0.8074282196450389, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:38:43,360] Trial 422 finished with value: 0.13084752147957812 and parameters: {'colsample_bylevel': 0.08982747415446041, 'learning_rate': 0.078997359782826, '

[I 2023-06-15 14:39:16,454] Trial 440 finished with value: 0.133692961393952 and parameters: {'colsample_bylevel': 0.07370486983299596, 'learning_rate': 0.07887051487199487, 'depth': 1, 'l2_leaf_reg': 1.8934188281201507, 'subsample': 0.47951411997341237, 'bagging_temperature': 3.774967368028593, 'model_size_reg': 0.29490959218667434, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:39:17,684] Trial 441 finished with value: 0.12936991989741492 and parameters: {'colsample_bylevel': 0.04267764218960432, 'learning_rate': 0.07530279173833845, 'depth': 2, 'l2_leaf_reg': 4.543168340887639, 'subsample': 0.308998934837554, 'bagging_temperature': 2.658600687347874, 'model_size_reg': 0.2728663168770507, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:39:19,432] Trial 442 finished with value: 0.12855753984147253 and parameters: {'colsample_bylevel': 0.07501632507811501, 'learning_rate': 0.07302509754077681

[I 2023-06-15 14:40:08,854] Trial 460 finished with value: 0.13652993960908505 and parameters: {'colsample_bylevel': 0.03659310533469003, 'learning_rate': 0.07814345671694872, 'depth': 2, 'l2_leaf_reg': 12.156324143933196, 'subsample': 0.1906102479412103, 'bagging_temperature': 7.728990635137931, 'model_size_reg': 0.2562387061602736, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:40:09,927] Trial 461 finished with value: 0.13261825873755526 and parameters: {'colsample_bylevel': 0.0211270510240288, 'learning_rate': 0.07387959310378878, 'depth': 3, 'l2_leaf_reg': 8.88536213632418, 'subsample': 0.27955010319328444, 'bagging_temperature': 4.512946778813094, 'model_size_reg': 0.3259500702890153, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:40:10,807] Trial 462 finished with value: 0.13478101696316586 and parameters: {'colsample_bylevel': 0.06277249662608499, 'learning_rate': 0.09687606012624406

[I 2023-06-15 14:40:35,497] Trial 480 finished with value: 0.1372278888842361 and parameters: {'colsample_bylevel': 0.04266377245373288, 'learning_rate': 0.07479782056679656, 'depth': 8, 'l2_leaf_reg': 0.7803042173140646, 'subsample': 0.23361744132402484, 'bagging_temperature': 4.129084211800133, 'model_size_reg': 0.4987889443581116, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:40:36,799] Trial 481 finished with value: 0.12563436136821798 and parameters: {'colsample_bylevel': 0.06999662008896068, 'learning_rate': 0.07150179777366394, 'depth': 3, 'l2_leaf_reg': 0.10295470960998365, 'subsample': 0.28732780594495566, 'bagging_temperature': 8.666210267479832, 'model_size_reg': 0.487956111201234, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:40:38,110] Trial 482 finished with value: 0.127999295249817 and parameters: {'colsample_bylevel': 0.0529323787527275, 'learning_rate': 0.09279914935411591

[I 2023-06-15 14:41:12,891] Trial 500 finished with value: 0.1346926001653438 and parameters: {'colsample_bylevel': 0.04114641508642349, 'learning_rate': 0.05421942551114503, 'depth': 2, 'l2_leaf_reg': 11.681132505614222, 'subsample': 0.2952765012782537, 'bagging_temperature': 8.157117845214172, 'model_size_reg': 0.48226747118947394, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:41:14,039] Trial 501 finished with value: 0.13040379591600837 and parameters: {'colsample_bylevel': 0.08302459127400272, 'learning_rate': 0.07367460929318632, 'depth': 2, 'l2_leaf_reg': 0.5546575912667145, 'subsample': 0.17610106415461957, 'bagging_temperature': 5.466447110936805, 'model_size_reg': 0.9527868121392176, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:41:15,464] Trial 502 finished with value: 0.1439919233573037 and parameters: {'colsample_bylevel': 0.05743934022816556, 'learning_rate': 0.011520323966115

[I 2023-06-15 14:41:40,633] Trial 520 finished with value: 0.13191272039988033 and parameters: {'colsample_bylevel': 0.0822488318062313, 'learning_rate': 0.0798148238015799, 'depth': 2, 'l2_leaf_reg': 2.619702901862926, 'subsample': 0.22150022696722932, 'bagging_temperature': 5.006232292705862, 'model_size_reg': 0.24265717825685287, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:41:41,800] Trial 521 finished with value: 0.12964478944115476 and parameters: {'colsample_bylevel': 0.06588365692832322, 'learning_rate': 0.09989895125064102, 'depth': 2, 'l2_leaf_reg': 9.19401077462305, 'subsample': 0.7491602101156424, 'bagging_temperature': 8.98793877688199, 'model_size_reg': 0.22647691526615638, 'boosting_type': 'Plain'}. Best is trial 363 with value: 0.12368272383839385.
[I 2023-06-15 14:41:43,445] Trial 522 finished with value: 0.12604329469498388 and parameters: {'colsample_bylevel': 0.05858094917354193, 'learning_rate': 0.09509439371512561,

[I 2023-06-15 14:42:15,727] Trial 540 finished with value: 0.12683455033719546 and parameters: {'colsample_bylevel': 0.08621926716574714, 'learning_rate': 0.06783919673948681, 'depth': 3, 'l2_leaf_reg': 12.937191352211626, 'subsample': 0.6168587955543429, 'bagging_temperature': 9.512229791728052, 'model_size_reg': 0.30120054767546056, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:42:17,109] Trial 541 finished with value: 0.13045031870634763 and parameters: {'colsample_bylevel': 0.045724304118242805, 'learning_rate': 0.03635588088834812, 'depth': 3, 'l2_leaf_reg': 5.57322857687988, 'subsample': 0.31288827771944144, 'bagging_temperature': 9.820318524112654, 'model_size_reg': 0.8820353061705779, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:42:18,249] Trial 542 finished with value: 0.1355223871118383 and parameters: {'colsample_bylevel': 0.03847656482891036, 'learning_rate': 0.050852714255450

[I 2023-06-15 14:42:56,028] Trial 560 finished with value: 0.13336387924529136 and parameters: {'colsample_bylevel': 0.03293350364589989, 'learning_rate': 0.08175301969218865, 'depth': 6, 'l2_leaf_reg': 3.8174861013058496, 'subsample': 0.7810291330546687, 'bagging_temperature': 9.144580669022103, 'model_size_reg': 0.36763456169214687, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:42:56,910] Trial 561 finished with value: 0.13436881614639337 and parameters: {'colsample_bylevel': 0.08476869608148539, 'learning_rate': 0.03826495662449553, 'depth': 1, 'l2_leaf_reg': 0.6136722932938254, 'subsample': 0.2658684840324586, 'bagging_temperature': 9.62721651776533, 'model_size_reg': 0.33875194920046836, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:42:58,291] Trial 562 finished with value: 0.13093121201077446 and parameters: {'colsample_bylevel': 0.041963136036300766, 'learning_rate': 0.0756780322591

[I 2023-06-15 14:43:22,402] Trial 580 finished with value: 0.12831473336361846 and parameters: {'colsample_bylevel': 0.056382609874577695, 'learning_rate': 0.08424451650400387, 'depth': 2, 'l2_leaf_reg': 7.060933692906376, 'subsample': 0.6472145927807014, 'bagging_temperature': 4.311520157640375, 'model_size_reg': 0.712822012637604, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:43:23,669] Trial 581 finished with value: 0.12869865637414524 and parameters: {'colsample_bylevel': 0.061264255489679, 'learning_rate': 0.09126340796974497, 'depth': 2, 'l2_leaf_reg': 6.753348933233541, 'subsample': 0.6992886913161989, 'bagging_temperature': 4.683743968854606, 'model_size_reg': 0.6929560258147569, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:43:24,770] Trial 582 finished with value: 0.13295378749208353 and parameters: {'colsample_bylevel': 0.05392905556820581, 'learning_rate': 0.08813119212167905, 

[I 2023-06-15 14:44:12,187] Trial 600 finished with value: 0.13386848378166316 and parameters: {'colsample_bylevel': 0.016417206426928586, 'learning_rate': 0.08145319816419905, 'depth': 2, 'l2_leaf_reg': 4.809385867461415, 'subsample': 0.6859491003460392, 'bagging_temperature': 3.039909126356732, 'model_size_reg': 0.7060070348142674, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:44:13,628] Trial 601 finished with value: 0.13195184108234048 and parameters: {'colsample_bylevel': 0.0636713382431368, 'learning_rate': 0.04761370511727668, 'depth': 2, 'l2_leaf_reg': 5.227360856173512, 'subsample': 0.619755973544483, 'bagging_temperature': 0.3610144948779124, 'model_size_reg': 0.3346872464186746, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:44:15,354] Trial 602 finished with value: 0.13135892503612975 and parameters: {'colsample_bylevel': 0.0551098200529895, 'learning_rate': 0.05276576266196587,

[I 2023-06-15 14:44:57,642] Trial 620 finished with value: 0.129719532846453 and parameters: {'colsample_bylevel': 0.03611246036371492, 'learning_rate': 0.0771580642655626, 'depth': 4, 'l2_leaf_reg': 4.844712528544823, 'subsample': 0.9280411960938635, 'bagging_temperature': 6.320340854713531, 'model_size_reg': 0.5088092703367679, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:44:58,579] Trial 621 finished with value: 0.13304885691068036 and parameters: {'colsample_bylevel': 0.05784761000140591, 'learning_rate': 0.09391640094442066, 'depth': 1, 'l2_leaf_reg': 1.0696568037962912, 'subsample': 0.7449945051862795, 'bagging_temperature': 6.6459798257327245, 'model_size_reg': 0.991500769306353, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:45:01,178] Trial 622 finished with value: 0.1353078861404536 and parameters: {'colsample_bylevel': 0.0697361550989474, 'learning_rate': 0.08977699950352082, 'd

[I 2023-06-15 14:45:35,715] Trial 640 finished with value: 0.14651037728268929 and parameters: {'colsample_bylevel': 0.015393637268217162, 'learning_rate': 0.08399997186001462, 'depth': 1, 'l2_leaf_reg': 23.775636551767995, 'subsample': 0.32691405459303247, 'bagging_temperature': 4.644179452511574, 'model_size_reg': 0.897486311919898, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:45:38,856] Trial 641 finished with value: 0.1337253738544827 and parameters: {'colsample_bylevel': 0.08580899825295893, 'learning_rate': 0.0884005256587934, 'depth': 3, 'l2_leaf_reg': 21.631780632018085, 'subsample': 0.7135273246029361, 'bagging_temperature': 8.917245795687364, 'model_size_reg': 0.2942581447923691, 'boosting_type': 'Ordered'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:45:39,908] Trial 642 finished with value: 0.1288743945985206 and parameters: {'colsample_bylevel': 0.04621632073930919, 'learning_rate': 0.074956248625320

[I 2023-06-15 14:46:04,435] Trial 660 finished with value: 0.13876472962038933 and parameters: {'colsample_bylevel': 0.056147799202797576, 'learning_rate': 0.06365510585984022, 'depth': 2, 'l2_leaf_reg': 20.344137313843518, 'subsample': 0.4854523363581111, 'bagging_temperature': 8.755629829584644, 'model_size_reg': 0.5764782094725281, 'boosting_type': 'Ordered'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:46:05,565] Trial 661 finished with value: 0.1291015069879821 and parameters: {'colsample_bylevel': 0.05898421456770897, 'learning_rate': 0.07493072913798636, 'depth': 2, 'l2_leaf_reg': 5.531958210913477, 'subsample': 0.7490893157820153, 'bagging_temperature': 9.011310785418203, 'model_size_reg': 0.2317686938536118, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:46:07,119] Trial 662 finished with value: 0.12916679274661655 and parameters: {'colsample_bylevel': 0.08384562426104772, 'learning_rate': 0.05052075169400

[I 2023-06-15 14:46:48,863] Trial 680 finished with value: 0.13007166811872103 and parameters: {'colsample_bylevel': 0.07850750593100041, 'learning_rate': 0.0749917019252597, 'depth': 3, 'l2_leaf_reg': 6.9415944936948595, 'subsample': 0.1801118377810139, 'bagging_temperature': 5.108440277931112, 'model_size_reg': 0.30330296143927343, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:46:50,842] Trial 681 finished with value: 0.13092556196018115 and parameters: {'colsample_bylevel': 0.04147314724782587, 'learning_rate': 0.09153510035217083, 'depth': 3, 'l2_leaf_reg': 2.976057985999568, 'subsample': 0.5436273203877795, 'bagging_temperature': 4.464917958323956, 'model_size_reg': 0.4570016071872297, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:46:53,131] Trial 682 finished with value: 0.12732954275487648 and parameters: {'colsample_bylevel': 0.08959763276021832, 'learning_rate': 0.0721738680023694

[I 2023-06-15 14:47:31,328] Trial 700 finished with value: 0.1328807402130188 and parameters: {'colsample_bylevel': 0.05325496619451585, 'learning_rate': 0.049122446342118146, 'depth': 2, 'l2_leaf_reg': 16.067407369146153, 'subsample': 0.9645616228786331, 'bagging_temperature': 2.282079839082057, 'model_size_reg': 0.17568999140578995, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:47:32,709] Trial 701 finished with value: 0.1333122365004184 and parameters: {'colsample_bylevel': 0.0394030387283081, 'learning_rate': 0.07229282817136476, 'depth': 2, 'l2_leaf_reg': 18.190217419189803, 'subsample': 0.7831570628959589, 'bagging_temperature': 5.962214950234117, 'model_size_reg': 0.9506734728380908, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:47:34,118] Trial 702 finished with value: 0.13561846928705182 and parameters: {'colsample_bylevel': 0.04956672303731348, 'learning_rate': 0.0249850564321671

[I 2023-06-15 14:48:01,599] Trial 720 finished with value: 0.12753605183061784 and parameters: {'colsample_bylevel': 0.05941453759514873, 'learning_rate': 0.0740498802755282, 'depth': 3, 'l2_leaf_reg': 1.781919776336462, 'subsample': 0.6286235504495673, 'bagging_temperature': 2.5970197096071934, 'model_size_reg': 0.8266466893582592, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:48:03,333] Trial 721 finished with value: 0.13184278734408328 and parameters: {'colsample_bylevel': 0.06968646540472968, 'learning_rate': 0.0768875018787629, 'depth': 2, 'l2_leaf_reg': 15.02033917813279, 'subsample': 0.22256528920406177, 'bagging_temperature': 6.774417814757927, 'model_size_reg': 0.7379259896391677, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:48:05,389] Trial 722 finished with value: 0.13073085944998492 and parameters: {'colsample_bylevel': 0.051673567800003144, 'learning_rate': 0.0584940328633938

[I 2023-06-15 14:48:33,364] Trial 740 finished with value: 0.12667307725656093 and parameters: {'colsample_bylevel': 0.03874224259141236, 'learning_rate': 0.07372996324734092, 'depth': 3, 'l2_leaf_reg': 2.9387460179165172, 'subsample': 0.7009313975471843, 'bagging_temperature': 0.15128098074410068, 'model_size_reg': 0.714927843526683, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:48:34,791] Trial 741 finished with value: 0.13083654908064735 and parameters: {'colsample_bylevel': 0.05738047382810049, 'learning_rate': 0.08415155220726792, 'depth': 2, 'l2_leaf_reg': 21.532568214312384, 'subsample': 0.2986582869758009, 'bagging_temperature': 4.731578768179878, 'model_size_reg': 0.9627047771084482, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:48:36,350] Trial 742 finished with value: 0.15333344195658005 and parameters: {'colsample_bylevel': 0.07958612551275326, 'learning_rate': 0.00843151134502

[I 2023-06-15 14:50:09,653] Trial 760 finished with value: 0.12800711117711905 and parameters: {'colsample_bylevel': 0.0442397098401892, 'learning_rate': 0.07169412642140197, 'depth': 3, 'l2_leaf_reg': 5.8792237953359345, 'subsample': 0.886264673568641, 'bagging_temperature': 7.23996856005405, 'model_size_reg': 0.36539134856836875, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:50:11,399] Trial 761 finished with value: 0.13097044164729513 and parameters: {'colsample_bylevel': 0.052435892719543506, 'learning_rate': 0.061151401503700846, 'depth': 4, 'l2_leaf_reg': 0.3678549061311618, 'subsample': 0.29782072586393094, 'bagging_temperature': 5.87950275519073, 'model_size_reg': 0.12627231743754935, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:50:12,915] Trial 762 finished with value: 0.13020270022929475 and parameters: {'colsample_bylevel': 0.05626193734479044, 'learning_rate': 0.05159474592329

[I 2023-06-15 14:50:44,122] Trial 780 finished with value: 0.12743553570644356 and parameters: {'colsample_bylevel': 0.07222376033049288, 'learning_rate': 0.09183440057252543, 'depth': 3, 'l2_leaf_reg': 3.9230090533078705, 'subsample': 0.8677168697877016, 'bagging_temperature': 7.825477629848357, 'model_size_reg': 0.7050892462559644, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:50:45,613] Trial 781 finished with value: 0.14047129019372395 and parameters: {'colsample_bylevel': 0.08258999010156964, 'learning_rate': 0.0210994833820841, 'depth': 2, 'l2_leaf_reg': 24.143265869980013, 'subsample': 0.5116946208731294, 'bagging_temperature': 9.672669911710797, 'model_size_reg': 0.15579468708816843, 'boosting_type': 'Plain'}. Best is trial 528 with value: 0.12324749357375206.
[I 2023-06-15 14:50:46,847] Trial 782 finished with value: 0.13103568032775267 and parameters: {'colsample_bylevel': 0.06882141083360468, 'learning_rate': 0.072492385869364

KeyboardInterrupt: 

In [70]:
cat_model = CatBoostRegressor(colsample_bylevel=0.08004892886595912,
                                      learning_rate=0.08724584964954875,
                                      depth=3,
                                      l2_leaf_reg=20.46786447675221,
                                      subsample=0.8203440457879617,
                                      bagging_temperature=4.50221817725758,
                                      model_size_reg=0.9722196114116113,
                                      boosting_type='Plain')

In [71]:
X,y = shuffle(X,y, random_state=42)
X = X.reset_index(drop=True)
y = y.reset_index(drop=True)

In [72]:
def rmse_cv(model,X,y):
    rmse = np.sqrt(-cross_val_score(model, X, y, scoring="neg_mean_squared_error", cv=6))
    return rmse

In [73]:
print(rmse_cv(cat_model,X,y).mean())

0:	learn: 0.3757329	total: 1.08ms	remaining: 1.08s
1:	learn: 0.3611354	total: 2.32ms	remaining: 1.16s
2:	learn: 0.3474503	total: 3.17ms	remaining: 1.05s
3:	learn: 0.3339523	total: 4.04ms	remaining: 1s
4:	learn: 0.3209996	total: 4.8ms	remaining: 955ms
5:	learn: 0.3112332	total: 5.59ms	remaining: 926ms
6:	learn: 0.2991561	total: 6.27ms	remaining: 890ms
7:	learn: 0.2893737	total: 6.94ms	remaining: 861ms
8:	learn: 0.2788741	total: 7.7ms	remaining: 848ms
9:	learn: 0.2700166	total: 8.47ms	remaining: 839ms
10:	learn: 0.2622585	total: 9.31ms	remaining: 837ms
11:	learn: 0.2542313	total: 10.4ms	remaining: 855ms
12:	learn: 0.2489426	total: 11.4ms	remaining: 868ms
13:	learn: 0.2422726	total: 12.3ms	remaining: 863ms
14:	learn: 0.2364211	total: 13ms	remaining: 856ms
15:	learn: 0.2308106	total: 14ms	remaining: 861ms
16:	learn: 0.2266639	total: 14.9ms	remaining: 860ms
17:	learn: 0.2215121	total: 15.7ms	remaining: 855ms
18:	learn: 0.2180605	total: 16.5ms	remaining: 853ms
19:	learn: 0.2124035	total: 17.

235:	learn: 0.1090778	total: 186ms	remaining: 603ms
236:	learn: 0.1090216	total: 187ms	remaining: 603ms
237:	learn: 0.1086653	total: 188ms	remaining: 603ms
238:	learn: 0.1086299	total: 189ms	remaining: 601ms
239:	learn: 0.1085729	total: 190ms	remaining: 600ms
240:	learn: 0.1084345	total: 190ms	remaining: 599ms
241:	learn: 0.1084078	total: 191ms	remaining: 598ms
242:	learn: 0.1083686	total: 192ms	remaining: 597ms
243:	learn: 0.1083503	total: 192ms	remaining: 596ms
244:	learn: 0.1082975	total: 193ms	remaining: 595ms
245:	learn: 0.1081499	total: 194ms	remaining: 594ms
246:	learn: 0.1080482	total: 194ms	remaining: 593ms
247:	learn: 0.1080015	total: 195ms	remaining: 592ms
248:	learn: 0.1078617	total: 196ms	remaining: 591ms
249:	learn: 0.1078260	total: 197ms	remaining: 590ms
250:	learn: 0.1078149	total: 197ms	remaining: 589ms
251:	learn: 0.1077936	total: 198ms	remaining: 588ms
252:	learn: 0.1076186	total: 199ms	remaining: 587ms
253:	learn: 0.1075695	total: 200ms	remaining: 587ms
254:	learn: 

466:	learn: 0.0959622	total: 365ms	remaining: 416ms
467:	learn: 0.0959346	total: 366ms	remaining: 416ms
468:	learn: 0.0958584	total: 366ms	remaining: 415ms
469:	learn: 0.0958034	total: 367ms	remaining: 414ms
470:	learn: 0.0957502	total: 368ms	remaining: 413ms
471:	learn: 0.0957418	total: 368ms	remaining: 412ms
472:	learn: 0.0957308	total: 369ms	remaining: 411ms
473:	learn: 0.0956949	total: 370ms	remaining: 410ms
474:	learn: 0.0956840	total: 370ms	remaining: 409ms
475:	learn: 0.0956722	total: 371ms	remaining: 408ms
476:	learn: 0.0956674	total: 372ms	remaining: 408ms
477:	learn: 0.0956145	total: 373ms	remaining: 407ms
478:	learn: 0.0956091	total: 373ms	remaining: 406ms
479:	learn: 0.0955895	total: 374ms	remaining: 405ms
480:	learn: 0.0955854	total: 374ms	remaining: 404ms
481:	learn: 0.0955793	total: 375ms	remaining: 403ms
482:	learn: 0.0955581	total: 376ms	remaining: 402ms
483:	learn: 0.0955484	total: 377ms	remaining: 401ms
484:	learn: 0.0954425	total: 377ms	remaining: 401ms
485:	learn: 

707:	learn: 0.0881377	total: 545ms	remaining: 225ms
708:	learn: 0.0881306	total: 546ms	remaining: 224ms
709:	learn: 0.0880952	total: 547ms	remaining: 223ms
710:	learn: 0.0880916	total: 548ms	remaining: 223ms
711:	learn: 0.0880481	total: 548ms	remaining: 222ms
712:	learn: 0.0880342	total: 549ms	remaining: 221ms
713:	learn: 0.0880176	total: 550ms	remaining: 220ms
714:	learn: 0.0879524	total: 551ms	remaining: 219ms
715:	learn: 0.0879063	total: 551ms	remaining: 219ms
716:	learn: 0.0878717	total: 552ms	remaining: 218ms
717:	learn: 0.0878613	total: 553ms	remaining: 217ms
718:	learn: 0.0878061	total: 553ms	remaining: 216ms
719:	learn: 0.0878005	total: 554ms	remaining: 215ms
720:	learn: 0.0877284	total: 555ms	remaining: 215ms
721:	learn: 0.0876812	total: 556ms	remaining: 214ms
722:	learn: 0.0876744	total: 556ms	remaining: 213ms
723:	learn: 0.0876695	total: 557ms	remaining: 212ms
724:	learn: 0.0876412	total: 558ms	remaining: 212ms
725:	learn: 0.0875659	total: 558ms	remaining: 211ms
726:	learn: 

949:	learn: 0.0819338	total: 725ms	remaining: 38.2ms
950:	learn: 0.0819163	total: 726ms	remaining: 37.4ms
951:	learn: 0.0818832	total: 727ms	remaining: 36.7ms
952:	learn: 0.0818804	total: 728ms	remaining: 35.9ms
953:	learn: 0.0818751	total: 729ms	remaining: 35.1ms
954:	learn: 0.0818588	total: 729ms	remaining: 34.4ms
955:	learn: 0.0818548	total: 730ms	remaining: 33.6ms
956:	learn: 0.0818501	total: 731ms	remaining: 32.8ms
957:	learn: 0.0818122	total: 732ms	remaining: 32.1ms
958:	learn: 0.0818064	total: 732ms	remaining: 31.3ms
959:	learn: 0.0817780	total: 733ms	remaining: 30.5ms
960:	learn: 0.0817045	total: 734ms	remaining: 29.8ms
961:	learn: 0.0816534	total: 735ms	remaining: 29ms
962:	learn: 0.0816448	total: 735ms	remaining: 28.3ms
963:	learn: 0.0816408	total: 736ms	remaining: 27.5ms
964:	learn: 0.0816369	total: 737ms	remaining: 26.7ms
965:	learn: 0.0816193	total: 737ms	remaining: 26ms
966:	learn: 0.0816019	total: 738ms	remaining: 25.2ms
967:	learn: 0.0815988	total: 739ms	remaining: 24.4

151:	learn: 0.1183079	total: 109ms	remaining: 607ms
152:	learn: 0.1182607	total: 110ms	remaining: 607ms
153:	learn: 0.1181258	total: 111ms	remaining: 607ms
154:	learn: 0.1179595	total: 111ms	remaining: 607ms
155:	learn: 0.1177166	total: 112ms	remaining: 606ms
156:	learn: 0.1176438	total: 113ms	remaining: 606ms
157:	learn: 0.1174230	total: 113ms	remaining: 605ms
158:	learn: 0.1172876	total: 114ms	remaining: 605ms
159:	learn: 0.1170811	total: 115ms	remaining: 604ms
160:	learn: 0.1169967	total: 116ms	remaining: 603ms
161:	learn: 0.1167166	total: 116ms	remaining: 602ms
162:	learn: 0.1165870	total: 117ms	remaining: 602ms
163:	learn: 0.1163006	total: 118ms	remaining: 602ms
164:	learn: 0.1160921	total: 119ms	remaining: 601ms
165:	learn: 0.1158910	total: 119ms	remaining: 600ms
166:	learn: 0.1157581	total: 120ms	remaining: 600ms
167:	learn: 0.1156516	total: 121ms	remaining: 599ms
168:	learn: 0.1156083	total: 122ms	remaining: 598ms
169:	learn: 0.1154232	total: 123ms	remaining: 599ms
170:	learn: 

397:	learn: 0.0990205	total: 289ms	remaining: 437ms
398:	learn: 0.0990092	total: 289ms	remaining: 436ms
399:	learn: 0.0989992	total: 290ms	remaining: 435ms
400:	learn: 0.0989717	total: 291ms	remaining: 435ms
401:	learn: 0.0989022	total: 292ms	remaining: 434ms
402:	learn: 0.0988929	total: 292ms	remaining: 433ms
403:	learn: 0.0988779	total: 293ms	remaining: 432ms
404:	learn: 0.0988691	total: 294ms	remaining: 431ms
405:	learn: 0.0988530	total: 294ms	remaining: 431ms
406:	learn: 0.0987526	total: 295ms	remaining: 430ms
407:	learn: 0.0985837	total: 296ms	remaining: 429ms
408:	learn: 0.0985480	total: 296ms	remaining: 428ms
409:	learn: 0.0985334	total: 297ms	remaining: 428ms
410:	learn: 0.0985251	total: 298ms	remaining: 427ms
411:	learn: 0.0984347	total: 298ms	remaining: 426ms
412:	learn: 0.0984169	total: 299ms	remaining: 425ms
413:	learn: 0.0983192	total: 300ms	remaining: 424ms
414:	learn: 0.0981479	total: 301ms	remaining: 424ms
415:	learn: 0.0981246	total: 301ms	remaining: 423ms
416:	learn: 

673:	learn: 0.0885874	total: 469ms	remaining: 227ms
674:	learn: 0.0885300	total: 470ms	remaining: 226ms
675:	learn: 0.0885129	total: 471ms	remaining: 226ms
676:	learn: 0.0884579	total: 472ms	remaining: 225ms
677:	learn: 0.0884444	total: 473ms	remaining: 224ms
678:	learn: 0.0884121	total: 473ms	remaining: 224ms
679:	learn: 0.0883793	total: 474ms	remaining: 223ms
680:	learn: 0.0883753	total: 474ms	remaining: 222ms
681:	learn: 0.0882944	total: 475ms	remaining: 222ms
682:	learn: 0.0882881	total: 476ms	remaining: 221ms
683:	learn: 0.0882790	total: 477ms	remaining: 220ms
684:	learn: 0.0882687	total: 477ms	remaining: 219ms
685:	learn: 0.0882488	total: 478ms	remaining: 219ms
686:	learn: 0.0881974	total: 479ms	remaining: 218ms
687:	learn: 0.0881626	total: 480ms	remaining: 218ms
688:	learn: 0.0880984	total: 480ms	remaining: 217ms
689:	learn: 0.0880889	total: 481ms	remaining: 216ms
690:	learn: 0.0880422	total: 482ms	remaining: 215ms
691:	learn: 0.0879889	total: 483ms	remaining: 215ms
692:	learn: 

908:	learn: 0.0816185	total: 646ms	remaining: 64.7ms
909:	learn: 0.0816141	total: 648ms	remaining: 64ms
910:	learn: 0.0815213	total: 649ms	remaining: 63.4ms
911:	learn: 0.0814943	total: 650ms	remaining: 62.7ms
912:	learn: 0.0814758	total: 651ms	remaining: 62ms
913:	learn: 0.0814718	total: 652ms	remaining: 61.4ms
914:	learn: 0.0814525	total: 653ms	remaining: 60.7ms
915:	learn: 0.0814103	total: 654ms	remaining: 60ms
916:	learn: 0.0813917	total: 655ms	remaining: 59.3ms
917:	learn: 0.0813714	total: 656ms	remaining: 58.6ms
918:	learn: 0.0813627	total: 657ms	remaining: 57.9ms
919:	learn: 0.0813324	total: 657ms	remaining: 57.2ms
920:	learn: 0.0813324	total: 658ms	remaining: 56.4ms
921:	learn: 0.0812552	total: 659ms	remaining: 55.8ms
922:	learn: 0.0812504	total: 660ms	remaining: 55.1ms
923:	learn: 0.0812446	total: 661ms	remaining: 54.4ms
924:	learn: 0.0812104	total: 662ms	remaining: 53.7ms
925:	learn: 0.0812021	total: 663ms	remaining: 53ms
926:	learn: 0.0811737	total: 665ms	remaining: 52.3ms
9

89:	learn: 0.1291077	total: 62.1ms	remaining: 628ms
90:	learn: 0.1286999	total: 63ms	remaining: 629ms
91:	learn: 0.1282068	total: 63.9ms	remaining: 631ms
92:	learn: 0.1278369	total: 64.6ms	remaining: 630ms
93:	learn: 0.1276257	total: 65.2ms	remaining: 629ms
94:	learn: 0.1272793	total: 65.8ms	remaining: 627ms
95:	learn: 0.1269477	total: 66.5ms	remaining: 626ms
96:	learn: 0.1267662	total: 67.2ms	remaining: 625ms
97:	learn: 0.1264534	total: 67.8ms	remaining: 624ms
98:	learn: 0.1260874	total: 68.4ms	remaining: 623ms
99:	learn: 0.1256643	total: 69ms	remaining: 621ms
100:	learn: 0.1254327	total: 69.6ms	remaining: 619ms
101:	learn: 0.1249738	total: 70.2ms	remaining: 618ms
102:	learn: 0.1247255	total: 70.9ms	remaining: 617ms
103:	learn: 0.1245871	total: 71.4ms	remaining: 615ms
104:	learn: 0.1243688	total: 72.1ms	remaining: 615ms
105:	learn: 0.1241858	total: 72.7ms	remaining: 614ms
106:	learn: 0.1237626	total: 73.5ms	remaining: 613ms
107:	learn: 0.1232565	total: 74.1ms	remaining: 612ms
108:	lea

360:	learn: 0.0983262	total: 243ms	remaining: 430ms
361:	learn: 0.0983115	total: 244ms	remaining: 430ms
362:	learn: 0.0981944	total: 245ms	remaining: 430ms
363:	learn: 0.0981806	total: 246ms	remaining: 429ms
364:	learn: 0.0981596	total: 246ms	remaining: 429ms
365:	learn: 0.0980934	total: 247ms	remaining: 428ms
366:	learn: 0.0980427	total: 248ms	remaining: 427ms
367:	learn: 0.0980308	total: 248ms	remaining: 426ms
368:	learn: 0.0979766	total: 249ms	remaining: 426ms
369:	learn: 0.0979100	total: 250ms	remaining: 425ms
370:	learn: 0.0978938	total: 250ms	remaining: 425ms
371:	learn: 0.0977969	total: 251ms	remaining: 424ms
372:	learn: 0.0976743	total: 252ms	remaining: 423ms
373:	learn: 0.0976281	total: 253ms	remaining: 423ms
374:	learn: 0.0976096	total: 253ms	remaining: 422ms
375:	learn: 0.0975453	total: 254ms	remaining: 422ms
376:	learn: 0.0975313	total: 255ms	remaining: 421ms
377:	learn: 0.0975139	total: 255ms	remaining: 420ms
378:	learn: 0.0974297	total: 256ms	remaining: 420ms
379:	learn: 

609:	learn: 0.0888469	total: 424ms	remaining: 271ms
610:	learn: 0.0887875	total: 425ms	remaining: 270ms
611:	learn: 0.0887663	total: 426ms	remaining: 270ms
612:	learn: 0.0887620	total: 426ms	remaining: 269ms
613:	learn: 0.0887467	total: 427ms	remaining: 269ms
614:	learn: 0.0887357	total: 428ms	remaining: 268ms
615:	learn: 0.0887171	total: 429ms	remaining: 267ms
616:	learn: 0.0887123	total: 429ms	remaining: 266ms
617:	learn: 0.0887068	total: 430ms	remaining: 266ms
618:	learn: 0.0886527	total: 431ms	remaining: 265ms
619:	learn: 0.0886141	total: 431ms	remaining: 264ms
620:	learn: 0.0886077	total: 432ms	remaining: 264ms
621:	learn: 0.0886001	total: 433ms	remaining: 263ms
622:	learn: 0.0885917	total: 433ms	remaining: 262ms
623:	learn: 0.0885875	total: 434ms	remaining: 261ms
624:	learn: 0.0885397	total: 434ms	remaining: 261ms
625:	learn: 0.0885083	total: 435ms	remaining: 260ms
626:	learn: 0.0885026	total: 436ms	remaining: 259ms
627:	learn: 0.0884359	total: 436ms	remaining: 259ms
628:	learn: 

875:	learn: 0.0820848	total: 605ms	remaining: 85.6ms
876:	learn: 0.0820231	total: 606ms	remaining: 85ms
877:	learn: 0.0820170	total: 607ms	remaining: 84.3ms
878:	learn: 0.0819939	total: 608ms	remaining: 83.6ms
879:	learn: 0.0819690	total: 608ms	remaining: 83ms
880:	learn: 0.0819521	total: 609ms	remaining: 82.3ms
881:	learn: 0.0819166	total: 610ms	remaining: 81.6ms
882:	learn: 0.0819126	total: 611ms	remaining: 80.9ms
883:	learn: 0.0819080	total: 611ms	remaining: 80.2ms
884:	learn: 0.0818960	total: 612ms	remaining: 79.5ms
885:	learn: 0.0818449	total: 613ms	remaining: 78.8ms
886:	learn: 0.0817898	total: 613ms	remaining: 78.1ms
887:	learn: 0.0816783	total: 614ms	remaining: 77.4ms
888:	learn: 0.0816707	total: 615ms	remaining: 76.8ms
889:	learn: 0.0816630	total: 615ms	remaining: 76.1ms
890:	learn: 0.0816309	total: 616ms	remaining: 75.4ms
891:	learn: 0.0815845	total: 617ms	remaining: 74.7ms
892:	learn: 0.0815478	total: 618ms	remaining: 74ms
893:	learn: 0.0815436	total: 618ms	remaining: 73.3ms

35:	learn: 0.1683866	total: 28.5ms	remaining: 762ms
36:	learn: 0.1666523	total: 29.5ms	remaining: 767ms
37:	learn: 0.1645446	total: 30.5ms	remaining: 773ms
38:	learn: 0.1626933	total: 31.3ms	remaining: 772ms
39:	learn: 0.1609148	total: 32.1ms	remaining: 770ms
40:	learn: 0.1599365	total: 32.8ms	remaining: 767ms
41:	learn: 0.1583728	total: 33.5ms	remaining: 763ms
42:	learn: 0.1568744	total: 34.2ms	remaining: 761ms
43:	learn: 0.1554486	total: 35ms	remaining: 761ms
44:	learn: 0.1545860	total: 35.7ms	remaining: 758ms
45:	learn: 0.1536018	total: 36.4ms	remaining: 755ms
46:	learn: 0.1525636	total: 37ms	remaining: 751ms
47:	learn: 0.1513781	total: 37.8ms	remaining: 749ms
48:	learn: 0.1503570	total: 38.6ms	remaining: 748ms
49:	learn: 0.1494845	total: 39.3ms	remaining: 747ms
50:	learn: 0.1484632	total: 40ms	remaining: 745ms
51:	learn: 0.1481256	total: 40.8ms	remaining: 744ms
52:	learn: 0.1471386	total: 41.5ms	remaining: 742ms
53:	learn: 0.1460540	total: 42.3ms	remaining: 741ms
54:	learn: 0.14465

278:	learn: 0.1048846	total: 208ms	remaining: 537ms
279:	learn: 0.1047710	total: 209ms	remaining: 537ms
280:	learn: 0.1046257	total: 210ms	remaining: 537ms
281:	learn: 0.1045436	total: 211ms	remaining: 536ms
282:	learn: 0.1045436	total: 211ms	remaining: 534ms
283:	learn: 0.1045363	total: 212ms	remaining: 533ms
284:	learn: 0.1044987	total: 212ms	remaining: 533ms
285:	learn: 0.1044479	total: 213ms	remaining: 532ms
286:	learn: 0.1042692	total: 214ms	remaining: 531ms
287:	learn: 0.1042486	total: 214ms	remaining: 530ms
288:	learn: 0.1041579	total: 215ms	remaining: 529ms
289:	learn: 0.1041265	total: 216ms	remaining: 528ms
290:	learn: 0.1040045	total: 217ms	remaining: 528ms
291:	learn: 0.1039813	total: 217ms	remaining: 527ms
292:	learn: 0.1039393	total: 218ms	remaining: 526ms
293:	learn: 0.1037942	total: 219ms	remaining: 525ms
294:	learn: 0.1037457	total: 220ms	remaining: 525ms
295:	learn: 0.1037253	total: 220ms	remaining: 524ms
296:	learn: 0.1036723	total: 221ms	remaining: 524ms
297:	learn: 

521:	learn: 0.0928589	total: 387ms	remaining: 354ms
522:	learn: 0.0928238	total: 388ms	remaining: 354ms
523:	learn: 0.0927584	total: 388ms	remaining: 353ms
524:	learn: 0.0927522	total: 389ms	remaining: 352ms
525:	learn: 0.0927073	total: 390ms	remaining: 351ms
526:	learn: 0.0926895	total: 391ms	remaining: 351ms
527:	learn: 0.0926835	total: 391ms	remaining: 350ms
528:	learn: 0.0926750	total: 392ms	remaining: 349ms
529:	learn: 0.0926655	total: 393ms	remaining: 348ms
530:	learn: 0.0926322	total: 393ms	remaining: 347ms
531:	learn: 0.0926113	total: 394ms	remaining: 347ms
532:	learn: 0.0925769	total: 395ms	remaining: 346ms
533:	learn: 0.0925544	total: 395ms	remaining: 345ms
534:	learn: 0.0925386	total: 396ms	remaining: 344ms
535:	learn: 0.0925342	total: 397ms	remaining: 343ms
536:	learn: 0.0924070	total: 397ms	remaining: 343ms
537:	learn: 0.0923970	total: 398ms	remaining: 342ms
538:	learn: 0.0923691	total: 399ms	remaining: 341ms
539:	learn: 0.0923599	total: 400ms	remaining: 340ms
540:	learn: 

785:	learn: 0.0856274	total: 567ms	remaining: 154ms
786:	learn: 0.0855888	total: 568ms	remaining: 154ms
787:	learn: 0.0855810	total: 569ms	remaining: 153ms
788:	learn: 0.0855724	total: 569ms	remaining: 152ms
789:	learn: 0.0855609	total: 570ms	remaining: 152ms
790:	learn: 0.0855436	total: 571ms	remaining: 151ms
791:	learn: 0.0855351	total: 571ms	remaining: 150ms
792:	learn: 0.0855172	total: 572ms	remaining: 149ms
793:	learn: 0.0855050	total: 573ms	remaining: 149ms
794:	learn: 0.0854919	total: 573ms	remaining: 148ms
795:	learn: 0.0854895	total: 574ms	remaining: 147ms
796:	learn: 0.0854717	total: 575ms	remaining: 146ms
797:	learn: 0.0854478	total: 575ms	remaining: 146ms
798:	learn: 0.0854414	total: 576ms	remaining: 145ms
799:	learn: 0.0854321	total: 577ms	remaining: 144ms
800:	learn: 0.0853633	total: 577ms	remaining: 143ms
801:	learn: 0.0852625	total: 578ms	remaining: 143ms
802:	learn: 0.0852459	total: 579ms	remaining: 142ms
803:	learn: 0.0852238	total: 580ms	remaining: 141ms
804:	learn: 

0:	learn: 0.3853769	total: 819us	remaining: 818ms
1:	learn: 0.3700944	total: 1.73ms	remaining: 861ms
2:	learn: 0.3540186	total: 2.31ms	remaining: 768ms
3:	learn: 0.3403329	total: 3.02ms	remaining: 753ms
4:	learn: 0.3269226	total: 3.79ms	remaining: 754ms
5:	learn: 0.3169750	total: 4.49ms	remaining: 744ms
6:	learn: 0.3046134	total: 5.05ms	remaining: 716ms
7:	learn: 0.2957130	total: 5.68ms	remaining: 704ms
8:	learn: 0.2849701	total: 6.38ms	remaining: 703ms
9:	learn: 0.2757583	total: 7.35ms	remaining: 728ms
10:	learn: 0.2676066	total: 8.22ms	remaining: 739ms
11:	learn: 0.2591514	total: 9.05ms	remaining: 745ms
12:	learn: 0.2539298	total: 9.9ms	remaining: 751ms
13:	learn: 0.2466784	total: 10.6ms	remaining: 749ms
14:	learn: 0.2406687	total: 11.4ms	remaining: 751ms
15:	learn: 0.2352665	total: 12.2ms	remaining: 752ms
16:	learn: 0.2305205	total: 12.9ms	remaining: 746ms
17:	learn: 0.2252433	total: 13.6ms	remaining: 741ms
18:	learn: 0.2216856	total: 14.2ms	remaining: 731ms
19:	learn: 0.2158702	tot

278:	learn: 0.1077028	total: 189ms	remaining: 489ms
279:	learn: 0.1075823	total: 190ms	remaining: 489ms
280:	learn: 0.1074254	total: 191ms	remaining: 488ms
281:	learn: 0.1073700	total: 192ms	remaining: 488ms
282:	learn: 0.1073700	total: 192ms	remaining: 486ms
283:	learn: 0.1073312	total: 193ms	remaining: 485ms
284:	learn: 0.1072984	total: 193ms	remaining: 485ms
285:	learn: 0.1072261	total: 194ms	remaining: 484ms
286:	learn: 0.1070414	total: 195ms	remaining: 483ms
287:	learn: 0.1070085	total: 195ms	remaining: 482ms
288:	learn: 0.1069136	total: 196ms	remaining: 481ms
289:	learn: 0.1068948	total: 196ms	remaining: 481ms
290:	learn: 0.1067914	total: 197ms	remaining: 480ms
291:	learn: 0.1067826	total: 198ms	remaining: 479ms
292:	learn: 0.1067458	total: 198ms	remaining: 478ms
293:	learn: 0.1065377	total: 199ms	remaining: 477ms
294:	learn: 0.1064957	total: 200ms	remaining: 477ms
295:	learn: 0.1064566	total: 200ms	remaining: 476ms
296:	learn: 0.1064170	total: 201ms	remaining: 475ms
297:	learn: 

549:	learn: 0.0937724	total: 370ms	remaining: 302ms
550:	learn: 0.0937618	total: 371ms	remaining: 302ms
551:	learn: 0.0937582	total: 371ms	remaining: 301ms
552:	learn: 0.0936491	total: 372ms	remaining: 301ms
553:	learn: 0.0936121	total: 373ms	remaining: 300ms
554:	learn: 0.0935970	total: 374ms	remaining: 299ms
555:	learn: 0.0935787	total: 374ms	remaining: 299ms
556:	learn: 0.0934704	total: 375ms	remaining: 298ms
557:	learn: 0.0934575	total: 375ms	remaining: 297ms
558:	learn: 0.0934255	total: 376ms	remaining: 297ms
559:	learn: 0.0934198	total: 377ms	remaining: 296ms
560:	learn: 0.0933519	total: 378ms	remaining: 296ms
561:	learn: 0.0932862	total: 378ms	remaining: 295ms
562:	learn: 0.0932767	total: 379ms	remaining: 294ms
563:	learn: 0.0932707	total: 380ms	remaining: 293ms
564:	learn: 0.0932520	total: 380ms	remaining: 293ms
565:	learn: 0.0932362	total: 381ms	remaining: 292ms
566:	learn: 0.0932190	total: 382ms	remaining: 292ms
567:	learn: 0.0932061	total: 382ms	remaining: 291ms
568:	learn: 

804:	learn: 0.0860640	total: 551ms	remaining: 133ms
805:	learn: 0.0860183	total: 552ms	remaining: 133ms
806:	learn: 0.0859513	total: 553ms	remaining: 132ms
807:	learn: 0.0858740	total: 553ms	remaining: 132ms
808:	learn: 0.0857950	total: 554ms	remaining: 131ms
809:	learn: 0.0857367	total: 555ms	remaining: 130ms
810:	learn: 0.0857246	total: 556ms	remaining: 129ms
811:	learn: 0.0857228	total: 556ms	remaining: 129ms
812:	learn: 0.0857065	total: 557ms	remaining: 128ms
813:	learn: 0.0856798	total: 558ms	remaining: 127ms
814:	learn: 0.0856565	total: 558ms	remaining: 127ms
815:	learn: 0.0856356	total: 559ms	remaining: 126ms
816:	learn: 0.0856119	total: 560ms	remaining: 125ms
817:	learn: 0.0856044	total: 560ms	remaining: 125ms
818:	learn: 0.0855978	total: 561ms	remaining: 124ms
819:	learn: 0.0855938	total: 562ms	remaining: 123ms
820:	learn: 0.0855883	total: 562ms	remaining: 123ms
821:	learn: 0.0855555	total: 563ms	remaining: 122ms
822:	learn: 0.0855509	total: 564ms	remaining: 121ms
823:	learn: 

3:	learn: 0.3537453	total: 4.1ms	remaining: 1.02s
4:	learn: 0.3390879	total: 5.2ms	remaining: 1.03s
5:	learn: 0.3301115	total: 6.14ms	remaining: 1.02s
6:	learn: 0.3188860	total: 7.29ms	remaining: 1.03s
7:	learn: 0.3080384	total: 8.06ms	remaining: 1000ms
8:	learn: 0.2998440	total: 8.74ms	remaining: 962ms
9:	learn: 0.2886128	total: 11.4ms	remaining: 1.13s
10:	learn: 0.2780102	total: 13.4ms	remaining: 1.2s
11:	learn: 0.2696499	total: 14.3ms	remaining: 1.18s
12:	learn: 0.2628988	total: 15.1ms	remaining: 1.15s
13:	learn: 0.2556730	total: 16.2ms	remaining: 1.14s
14:	learn: 0.2515174	total: 16.9ms	remaining: 1.11s
15:	learn: 0.2460410	total: 17.7ms	remaining: 1.08s
16:	learn: 0.2421776	total: 18.4ms	remaining: 1.06s
17:	learn: 0.2362955	total: 19.1ms	remaining: 1.04s
18:	learn: 0.2329741	total: 20.1ms	remaining: 1.03s
19:	learn: 0.2278039	total: 21.4ms	remaining: 1.05s
20:	learn: 0.2236789	total: 22.9ms	remaining: 1.07s
21:	learn: 0.2194995	total: 25.2ms	remaining: 1.12s
22:	learn: 0.2143943	

202:	learn: 0.1148767	total: 181ms	remaining: 712ms
203:	learn: 0.1147920	total: 182ms	remaining: 711ms
204:	learn: 0.1147446	total: 183ms	remaining: 711ms
205:	learn: 0.1146952	total: 184ms	remaining: 709ms
206:	learn: 0.1146125	total: 185ms	remaining: 707ms
207:	learn: 0.1145695	total: 185ms	remaining: 706ms
208:	learn: 0.1143846	total: 186ms	remaining: 703ms
209:	learn: 0.1142008	total: 187ms	remaining: 702ms
210:	learn: 0.1141234	total: 187ms	remaining: 700ms
211:	learn: 0.1141064	total: 188ms	remaining: 698ms
212:	learn: 0.1141062	total: 188ms	remaining: 695ms
213:	learn: 0.1140761	total: 189ms	remaining: 693ms
214:	learn: 0.1140106	total: 189ms	remaining: 692ms
215:	learn: 0.1139479	total: 190ms	remaining: 691ms
216:	learn: 0.1138953	total: 191ms	remaining: 689ms
217:	learn: 0.1138805	total: 192ms	remaining: 687ms
218:	learn: 0.1136997	total: 192ms	remaining: 685ms
219:	learn: 0.1136587	total: 193ms	remaining: 684ms
220:	learn: 0.1136226	total: 194ms	remaining: 682ms
221:	learn: 

462:	learn: 0.0994944	total: 362ms	remaining: 420ms
463:	learn: 0.0994753	total: 363ms	remaining: 419ms
464:	learn: 0.0994244	total: 364ms	remaining: 419ms
465:	learn: 0.0993702	total: 365ms	remaining: 418ms
466:	learn: 0.0993498	total: 365ms	remaining: 417ms
467:	learn: 0.0992947	total: 366ms	remaining: 416ms
468:	learn: 0.0992288	total: 367ms	remaining: 415ms
469:	learn: 0.0991976	total: 367ms	remaining: 414ms
470:	learn: 0.0991335	total: 368ms	remaining: 414ms
471:	learn: 0.0991253	total: 369ms	remaining: 413ms
472:	learn: 0.0989785	total: 370ms	remaining: 412ms
473:	learn: 0.0989552	total: 370ms	remaining: 411ms
474:	learn: 0.0989269	total: 371ms	remaining: 410ms
475:	learn: 0.0989135	total: 371ms	remaining: 409ms
476:	learn: 0.0988009	total: 372ms	remaining: 408ms
477:	learn: 0.0987580	total: 373ms	remaining: 407ms
478:	learn: 0.0986678	total: 373ms	remaining: 406ms
479:	learn: 0.0986423	total: 374ms	remaining: 405ms
480:	learn: 0.0986259	total: 374ms	remaining: 404ms
481:	learn: 

717:	learn: 0.0901264	total: 542ms	remaining: 213ms
718:	learn: 0.0900943	total: 543ms	remaining: 212ms
719:	learn: 0.0900873	total: 544ms	remaining: 212ms
720:	learn: 0.0900481	total: 545ms	remaining: 211ms
721:	learn: 0.0900367	total: 546ms	remaining: 210ms
722:	learn: 0.0900301	total: 546ms	remaining: 209ms
723:	learn: 0.0899794	total: 547ms	remaining: 209ms
724:	learn: 0.0899732	total: 548ms	remaining: 208ms
725:	learn: 0.0899493	total: 548ms	remaining: 207ms
726:	learn: 0.0899149	total: 549ms	remaining: 206ms
727:	learn: 0.0899120	total: 550ms	remaining: 205ms
728:	learn: 0.0899037	total: 550ms	remaining: 205ms
729:	learn: 0.0898394	total: 551ms	remaining: 204ms
730:	learn: 0.0898343	total: 552ms	remaining: 203ms
731:	learn: 0.0897885	total: 552ms	remaining: 202ms
732:	learn: 0.0897225	total: 553ms	remaining: 201ms
733:	learn: 0.0896942	total: 554ms	remaining: 201ms
734:	learn: 0.0896884	total: 554ms	remaining: 200ms
735:	learn: 0.0896613	total: 555ms	remaining: 199ms
736:	learn: 

959:	learn: 0.0835387	total: 724ms	remaining: 30.1ms
960:	learn: 0.0835243	total: 725ms	remaining: 29.4ms
961:	learn: 0.0834984	total: 725ms	remaining: 28.6ms
962:	learn: 0.0834909	total: 726ms	remaining: 27.9ms
963:	learn: 0.0834822	total: 727ms	remaining: 27.1ms
964:	learn: 0.0834793	total: 728ms	remaining: 26.4ms
965:	learn: 0.0834753	total: 728ms	remaining: 25.6ms
966:	learn: 0.0834399	total: 729ms	remaining: 24.9ms
967:	learn: 0.0834192	total: 730ms	remaining: 24.1ms
968:	learn: 0.0834147	total: 730ms	remaining: 23.4ms
969:	learn: 0.0833456	total: 731ms	remaining: 22.6ms
970:	learn: 0.0833242	total: 732ms	remaining: 21.9ms
971:	learn: 0.0832474	total: 733ms	remaining: 21.1ms
972:	learn: 0.0832373	total: 733ms	remaining: 20.3ms
973:	learn: 0.0832049	total: 734ms	remaining: 19.6ms
974:	learn: 0.0831877	total: 735ms	remaining: 18.8ms
975:	learn: 0.0831648	total: 735ms	remaining: 18.1ms
976:	learn: 0.0831289	total: 736ms	remaining: 17.3ms
977:	learn: 0.0831058	total: 737ms	remaining: 

In [74]:
submit = test[['Id']]
submit = submit.reset_index(drop=True)

In [75]:
cat_model.fit(X,y)

0:	learn: 0.3839057	total: 822us	remaining: 822ms
1:	learn: 0.3686327	total: 2.26ms	remaining: 1.13s
2:	learn: 0.3523335	total: 3.29ms	remaining: 1.09s
3:	learn: 0.3386238	total: 4.25ms	remaining: 1.06s
4:	learn: 0.3253388	total: 5.17ms	remaining: 1.03s
5:	learn: 0.3153386	total: 5.99ms	remaining: 992ms
6:	learn: 0.3029196	total: 6.74ms	remaining: 956ms
7:	learn: 0.2929963	total: 7.45ms	remaining: 924ms
8:	learn: 0.2827511	total: 8.24ms	remaining: 907ms
9:	learn: 0.2735166	total: 9.04ms	remaining: 895ms
10:	learn: 0.2654625	total: 9.91ms	remaining: 891ms
11:	learn: 0.2569053	total: 10.7ms	remaining: 877ms
12:	learn: 0.2516382	total: 11.5ms	remaining: 875ms
13:	learn: 0.2450370	total: 12.8ms	remaining: 900ms
14:	learn: 0.2389828	total: 13.8ms	remaining: 905ms
15:	learn: 0.2335393	total: 14.9ms	remaining: 916ms
16:	learn: 0.2287827	total: 16ms	remaining: 923ms
17:	learn: 0.2235564	total: 16.8ms	remaining: 918ms
18:	learn: 0.2201828	total: 17.8ms	remaining: 917ms
19:	learn: 0.2144423	tota

228:	learn: 0.1112684	total: 182ms	remaining: 612ms
229:	learn: 0.1111868	total: 183ms	remaining: 612ms
230:	learn: 0.1110638	total: 184ms	remaining: 612ms
231:	learn: 0.1109915	total: 184ms	remaining: 611ms
232:	learn: 0.1109628	total: 185ms	remaining: 610ms
233:	learn: 0.1108027	total: 186ms	remaining: 609ms
234:	learn: 0.1107482	total: 187ms	remaining: 608ms
235:	learn: 0.1106810	total: 187ms	remaining: 607ms
236:	learn: 0.1105571	total: 188ms	remaining: 606ms
237:	learn: 0.1103984	total: 189ms	remaining: 604ms
238:	learn: 0.1102696	total: 190ms	remaining: 603ms
239:	learn: 0.1101714	total: 190ms	remaining: 602ms
240:	learn: 0.1101601	total: 191ms	remaining: 601ms
241:	learn: 0.1099487	total: 192ms	remaining: 600ms
242:	learn: 0.1097987	total: 193ms	remaining: 600ms
243:	learn: 0.1097412	total: 193ms	remaining: 599ms
244:	learn: 0.1095580	total: 194ms	remaining: 598ms
245:	learn: 0.1095266	total: 195ms	remaining: 597ms
246:	learn: 0.1094817	total: 196ms	remaining: 597ms
247:	learn: 

452:	learn: 0.0985179	total: 363ms	remaining: 438ms
453:	learn: 0.0984756	total: 364ms	remaining: 437ms
454:	learn: 0.0984692	total: 365ms	remaining: 437ms
455:	learn: 0.0984597	total: 366ms	remaining: 436ms
456:	learn: 0.0984472	total: 366ms	remaining: 435ms
457:	learn: 0.0983342	total: 367ms	remaining: 434ms
458:	learn: 0.0983278	total: 368ms	remaining: 433ms
459:	learn: 0.0983102	total: 369ms	remaining: 433ms
460:	learn: 0.0983017	total: 369ms	remaining: 431ms
461:	learn: 0.0982802	total: 370ms	remaining: 431ms
462:	learn: 0.0982663	total: 371ms	remaining: 430ms
463:	learn: 0.0982438	total: 371ms	remaining: 429ms
464:	learn: 0.0982197	total: 372ms	remaining: 428ms
465:	learn: 0.0981487	total: 373ms	remaining: 427ms
466:	learn: 0.0981036	total: 374ms	remaining: 426ms
467:	learn: 0.0980698	total: 374ms	remaining: 425ms
468:	learn: 0.0980339	total: 375ms	remaining: 425ms
469:	learn: 0.0980030	total: 376ms	remaining: 424ms
470:	learn: 0.0979659	total: 377ms	remaining: 423ms
471:	learn: 

686:	learn: 0.0908868	total: 543ms	remaining: 248ms
687:	learn: 0.0908498	total: 544ms	remaining: 247ms
688:	learn: 0.0907963	total: 545ms	remaining: 246ms
689:	learn: 0.0907794	total: 546ms	remaining: 245ms
690:	learn: 0.0907773	total: 547ms	remaining: 245ms
691:	learn: 0.0907704	total: 548ms	remaining: 244ms
692:	learn: 0.0907116	total: 549ms	remaining: 243ms
693:	learn: 0.0907032	total: 549ms	remaining: 242ms
694:	learn: 0.0906705	total: 550ms	remaining: 241ms
695:	learn: 0.0906562	total: 551ms	remaining: 241ms
696:	learn: 0.0906230	total: 552ms	remaining: 240ms
697:	learn: 0.0905756	total: 553ms	remaining: 239ms
698:	learn: 0.0905714	total: 554ms	remaining: 238ms
699:	learn: 0.0905260	total: 555ms	remaining: 238ms
700:	learn: 0.0905241	total: 555ms	remaining: 237ms
701:	learn: 0.0904820	total: 556ms	remaining: 236ms
702:	learn: 0.0904735	total: 557ms	remaining: 235ms
703:	learn: 0.0904351	total: 558ms	remaining: 234ms
704:	learn: 0.0904206	total: 558ms	remaining: 234ms
705:	learn: 

915:	learn: 0.0851968	total: 726ms	remaining: 66.6ms
916:	learn: 0.0851832	total: 727ms	remaining: 65.8ms
917:	learn: 0.0851403	total: 728ms	remaining: 65ms
918:	learn: 0.0851237	total: 729ms	remaining: 64.2ms
919:	learn: 0.0851197	total: 729ms	remaining: 63.4ms
920:	learn: 0.0851197	total: 730ms	remaining: 62.6ms
921:	learn: 0.0850529	total: 730ms	remaining: 61.8ms
922:	learn: 0.0850490	total: 731ms	remaining: 61ms
923:	learn: 0.0850464	total: 732ms	remaining: 60.2ms
924:	learn: 0.0850413	total: 733ms	remaining: 59.4ms
925:	learn: 0.0850403	total: 733ms	remaining: 58.6ms
926:	learn: 0.0849862	total: 734ms	remaining: 57.8ms
927:	learn: 0.0849552	total: 735ms	remaining: 57ms
928:	learn: 0.0849328	total: 735ms	remaining: 56.2ms
929:	learn: 0.0849277	total: 736ms	remaining: 55.4ms
930:	learn: 0.0849241	total: 737ms	remaining: 54.6ms
931:	learn: 0.0848693	total: 738ms	remaining: 53.8ms
932:	learn: 0.0848539	total: 739ms	remaining: 53ms
933:	learn: 0.0848466	total: 739ms	remaining: 52.2ms
9

In [76]:
#submit_predict = cat_model.predict(test)
#submit_predict = np.exp(submit_predict)

In [77]:
submit['SalePrice'] = submit_predict

In [78]:
submit.to_csv('submit_optuna_catboost.csv', index=False)

Reducing feature skewness actually slightly increased the Kaggle score. 